### ISSUE TITLE
Error Android 7 

### ISSUE THREAD

**Issue by deycks** [gartnera/headunit#71]
When connecting my cell phone Samsung Galaxy S8 with Android 7 Nougat with Android Auto. Mazda Connect does not recognize it and does not start Andoid Auto.

My version of mazda firmware its 59.00.441

Thank you very much!

**Comment by viktorgino** [comment]
From the limited info you provided it sound like its the same issue I and others have experienced. I've created a fix for it, which has been merged here, but the binaries haven't been updated yet.
@dakoellis has compiled it with these changes included and uploaded it to Google Drive. You can get it [here](https://drive.google.com/open?id=0B3R02CIyKsApUlhYM0NNWkpIRzQ). To get it working uninstall the previous version and install this one. Please note that I don't actually own a Mazda, so I can't help with debugging that side.

**Comment by deycks** [comment]
Thanks a lot 
Before having the Galaxy S8 I used a Moto XPlay with Android 6.
Everything worked properly until i bought the Samsung.
On the phone, I installed Android Auto app, Enable debugging.

But when I started the Android Auto App on the Mazda and I connect the cell phone via USB.
Does not start


Do you think it's the same problem you have?


Thanks @viktorgino

**Comment by viktorgino** [comment]
@deycks This sound like the exact issue me and others were having. It worked on devices with USB 2 even with the latest version of Android, but not with the USB 3. I'd suggest giving it a go.

**Comment by deycks** [comment]
I install this version but, I got this error "_startContextReady: Can't connect to headunit process"

**Comment by viktorgino** [comment]
@deycks can you access the log?

**Comment by deycks** [comment]
@viktorgino this is?
Installer started.
MYDIR = /tmp/mnt/sda1
CMU version = 59.00.441A-NA
Check mount point ... OK
= POPUP: AA INSTALL SCRIPT Welcome to Android Auto installation script. Would you like to proceed? Proceed Abort
Disabling watchdog and remounting for write access ... SUCCESS
Check whether Android Auto is installed ... NO
Installing Android Auto.
= POPUP INFO: INSTALLING Android Auto is installing ...
Changing opera.ini file ... backup exists ... OK
Disabling fps.js ... FAILED
Changing stage_wifi.sh file ... backing up file first ... autostart entry added to /jci/scripts/stage_wifi.sh ... DONE
Copying AA files ... DONE
Changing permissions ... DONE


Running smoke test

/tmp/mnt/data_persist/dev/bin/headunit: line 1: syntax error: unexpected "("

Test went wrong.
= POPUP INFO: TEST RUN FAILED Headunit binary launch failed. Check the log.
Installation complete!
= POPUP CONFIRM: DONE Android Auto has been installed. System will reboot now. Remember to remove USB drive.

**Comment by dakoellis** [comment]
@deycks are you turning on the car before you plug in your phone?  Wait an hour to be safe before trying again, and make sure the car is all the way on (wait until you have control of the CMU) before you plug the phone in. I can usually plug it in as soon as the mazda sign shows, but I'd say wait longer just to make sure that's not the issue.

**Comment by deycks** [comment]
Yes the ignition, and I plug the Phone before I have the control os the CMU but the issue continues. 
But in this test I don't wait a hour (I wait about 20 minutes).  This is necesary?

**Comment by dakoellis** [comment]
@deycks It shouldn't be an issue but I became overly cautious since every once in a while a half hour wasn't enough for my car for some reason.  Try removing the one you have and installing this one: https://drive.google.com/open?id=0B3R02CIyKsApX2ZWOTFKOVNVNUU

**Comment by deycks** [comment]
@dakoellis :,,( the same error.
![20170731_212609](https://user-images.githubusercontent.com/30603472/28806744-47e545a6-7637-11e7-8ed9-2f23ef8279b1.jpg)


Installation Log
Installer started.
MYDIR = /tmp/mnt/sda1
CMU version = 59.00.441A-NA
Check mount point ... OK
= POPUP: AA INSTALL SCRIPT Welcome to Android Auto installation script. Would you like to proceed? Proceed Abort
Disabling watchdog and remounting for write access ... SUCCESS
Check whether Android Auto is installed ... NO
Installing Android Auto.
= POPUP INFO: INSTALLING Android Auto is installing ...
Changing opera.ini file ... backup exists ... OK
Disabling fps.js ... FAILED
Changing stage_wifi.sh file ... backup exists so leaving file as is - there will be no autostart. FAILED
Copying AA files ... DONE
Changing permissions ... DONE


Running smoke test

uname:
 sysname: Linux
 release: 3.0.35
 version: #1 SMP PREEMPT Sun Mar 15 10:12:10 IST 2015
 machine: armv7l
libprotoversion: 2.6.1
libusb_get_version:
 version: 1.0.9.0
 rc: 
 describe: 
openssl version: OpenSSL 1.0.2h  3 May 2016 (0x1000208f)

(gst-plugin-scanner:5101): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-0.10/libmfw_gst_mpeg2dec.so': lib_mpeg2_dec_arm11_elinux.so.2.2.0: cannot open shared object file: No such file or directory

(gst-plugin-scanner:5101): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-0.10/libmfw_gst_h264dec.so': lib_H264_dec_arm11_elinux.so.2.6.1: cannot open shared object file: No such file or directory

(gst-plugin-scanner:5101): GStreamer-CRITICAL **: gst_pad_template_new: assertion `caps != NULL' failed

(gst-plugin-scanner:5101): GStreamer-CRITICAL **: gst_element_class_add_pad_template: assertion `GST_IS_PAD_TEMPLATE (templ)' failed

(gst-plugin-scanner:5101): GStreamer-WARNING **: Failed to load plugin '/usr/lib/gstreamer-0.10/libmfw_gst_mpeg4aspdec.so': lib_MPEG4ASP_dec_arm11_elinux.so.1.2.1: cannot open shared object file: No such file or directory
[1;34mMAX resolution 800x480
[0m###TESTMODE_OK###

Test looks good.
Installation complete!
= POPUP CONFIRM: DONE Android Auto has been installed. System will reboot now. Remember to remove USB drive.

**Comment by dakoellis** [comment]
Just checked my installer log and that's exactly what mine looks like.  Only difference is my CMU version is 55.00.760A-NA.  That error is the one I was getting if I turn on my car too early before plugging it in (though I'm unable to reproduce it right now suspicously...) so maybe try going to the AA app before you plug in your phone and see if you already have that error?

**Comment by deycks** [comment]
@dakoellis Yes, if I start the AA app before plug my phone I see the same error

**Comment by dakoellis** [comment]
What version were you using before? Did you use the AIO to remove the old version or the zip file here?

**Comment by deycks** [comment]
First I remove the AA app with AIO, then I install the zip file of here.

**Comment by dakoellis** [comment]
@deycks all I got is trying uninstalling it and trying the old version. see if you still get the error on that version.  Maybe search around the car OS in ssh and see if you can find any error logs besides the installation one you posted before.  Sorry I don't know much of anything about the inner workings of the project.

**Comment by deycks** [comment]
:( 
thanks

**Comment by Greaseman85** [comment]
I have the same issue with my HTC U11, wouldn't work with the previous version, and the new one I installed gives me the same "Can't connect to headunit process" error.

**Comment by deycks** [comment]
I think the AA version of this page is the one that fails

**Comment by dakoellis** [comment]
Could be.  I could pull the binary I have in my car when I get home, but AFAIK the only difference I had with my car was the CMU version

**Comment by dakoellis** [comment]
Or maybe google drive is the issue... I'll upload it to another service

**Comment by deycks** [comment]
Ok thanks, and I Will try again

**Comment by deycks** [comment]
Yes please do that. 
Thanks

**Comment by dakoellis** [comment]
OK lets try this again.  I pulled the binary from my car, tgz instead of zip, and uploaded it to another service: https://ufile.io/gt6bf

**Comment by Greaseman85** [comment]
Works for me this time, thanks!  At first Google Play Services would crash when I plugged my phone in and would keep doing that a few times.  I then unplugged the phone, started the AA app in my headunit, then plugged the phone in, and it worked fine.  Now when I just plug the phone in it starts automatically like before.

**Comment by deycks** [comment]
Hi, excuse me. You first uninstall AS and then install the new binary or directly install the binary.
Thanks

**Comment by Greaseman85** [comment]
I uninstalled the old version before installing this one.

**Comment by trit0Ch** [comment]
I have one of the newest phones running a USB 3.0 cables, I plug my phone in and it doesn't recognize the usb port as anything other than a "charge" port. Could be one of the issues as I tested my older phone and it would get past the headunit credit screen into.... a totally black screen. But hey at least I got a reaction out of the CMU compared to my latest phone.

TL;DR
Moto Z2 Force USB-C 3.0 

- Cable not recognized by usb port other than charging
- Stuck on headunit credit screen

LG G4 micro USB

- Runs AA but Play services will crash
- Got beyond the headunit credit screen into total black screen instead

**Comment by Trevelopment** [comment]
From the comments here I think this should be closed

**Comment by roireshef** [comment]
@dakoellis did you notice you have uploaded that to a non-free hosting service? can you please upload it to somewhere that is free to download from?

**Comment by dakoellis** [comment]
No. It was just temporary to try to get it working. The main dev has updated it.
